### PR TITLE
trace_rawTransaction API

### PR DESCRIPTION
**Pull request by Sixtysixter**
Added trace_rawTransaction API implementation

### CODE REVIEW

File: silkrpc/commands/eth_api.cpp [hunk 1]
```diff
@@ -1229,6 +1229,8 @@ boost::asio::awaitable<void> EthereumRpcApi::handle_eth_call_bundle(const nlohma
                 break;
             }
 
+            SILKRPC_LOG << "diff_time: " << clock_time::since(start_time)/1000000 << " timeout: " << timeout << "\n";
```
Comment (canepat) [inline silkrpc/commands/eth_api.cpp:1232]: `SILKRPC_LOG` -> `SILKRPC_DEBUG`

File: silkrpc/commands/trace_api.cpp [hunk 2]
```diff
@@ -121,12 +121,77 @@ boost::asio::awaitable<void> TraceRpcApi::handle_trace_call_many(const nlohmann:
 
 // https://eth.wiki/json-rpc/API#trace_rawtransaction
 boost::asio::awaitable<void> TraceRpcApi::handle_trace_raw_transaction(const nlohmann::json& request, nlohmann::json& reply) {
+    auto params = request["params"];
```
Comment (canepat) [inline silkrpc/commands/trace_api.cpp:124]: `params` can be made `const` here

File: silkrpc/commands/trace_api.cpp [hunk 3]
```diff
@@ -121,12 +121,77 @@ boost::asio::awaitable<void> TraceRpcApi::handle_trace_call_many(const nlohmann:
...
+    if (params.size() < 2) {
+        auto error_msg = "invalid trace_rawTransaction params: " + params.dump();
```
Comment (canepat) [inline silkrpc/commands/trace_api.cpp:126]: `error_msg` can be made `const` here and below

File: silkrpc/commands/trace_api.cpp [hunk 4]
```diff
@@ -121,12 +121,77 @@ boost::asio::awaitable<void> TraceRpcApi::handle_trace_call_many(const nlohmann:
...
+
+    SILKRPC_INFO << "transaction: " << transaction << " config: " << config << "\n";
+
     auto tx = co_await database_->begin();
 
     try {
         ethdb::TransactionDatabase tx_database{*tx};
 
-        reply = make_json_error(request["id"], 500, "not yet implemented");
+        auto block_number = co_await core::get_latest_block_number(tx_database);
```
Comment (canepat) [inline silkrpc/commands/trace_api.cpp:183]: `block_number` can be made `const` here

**Review by canepat — CHANGES_REQUESTED** [review]

File: silkrpc/commands/trace_api.cpp (hunk 2 above)
Comment (Sixtysixter) [reply silkrpc/commands/trace_api.cpp:124]: Done

File: silkrpc/commands/eth_api.cpp (hunk 1 above)
Comment (Sixtysixter) [reply silkrpc/commands/eth_api.cpp:1232]: it was a typo

File: silkrpc/commands/trace_api.cpp (hunk 3 above)
Comment (Sixtysixter) [reply silkrpc/commands/trace_api.cpp:126]: done

File: silkrpc/commands/trace_api.cpp (hunk 4 above)
Comment (Sixtysixter) [reply silkrpc/commands/trace_api.cpp:183]: done

**Review by canepat — APPROVED** [review]

**Pull request merged**